owncloudData/src/test/java/com/owncloud/android/data/sharing/shares/repository/OCShareRepositoryTest.kt

```diff
@@ -20,496 +20,287 @@
+                permissions = permissions,
+                accountName = OC_ACCOUNT_NAME
             )
-        }
+        } returns OC_PUBLIC_SHARE
 
-        verify(exactly = 1) { localShareDataSource.update(shares.first()) }
-    }
-
-    @Test(expected = FileNotFoundException::class)
-    fun updatePublicShareFileNotFoundException() {
         every {
-            remoteShareDataSource.updateShare(
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any()
-            )
-        } throws FileNotFoundException()
+            localShareDataSource.update(OC_PUBLIC_SHARE)
+        } returns 1 // The result of this method is not used, so it can be anything
 
-        ocShareRepository.updatePublicShare(
-            OC_SHARE.remoteId,
-            "Docs link",
-            "password",
-            2000,
-            -1,
-            accountName
-        )
+        ocShareRepository.updatePublicShare(OC_PUBLIC_SHARE.remoteId, OC_PUBLIC_SHARE.name!!, password, expiration, permissions, OC_ACCOUNT_NAME)
 
         verify(exactly = 1) {
             remoteShareDataSource.updateShare(
-                OC_SHARE.remoteId,
-                "Docs link",
-                "password",
-                2000,
-                -1,
-                accountName
+                remoteId = OC_PUBLIC_SHARE.remoteId,
+                name = OC_PUBLIC_SHARE.name!!,
+                password = password,
+                expirationDateInMillis = expiration,
+                permissions = permissions,
+                accountName = OC_ACCOUNT_NAME
             )
-        }
-
-        verify(exactly = 0) {
-            localShareDataSource.update(shares.first())
+            localShareDataSource.update(OC_PUBLIC_SHARE)
         }
     }
 
     @Test
-    fun insertPrivateShareOk() {
+    fun `getSharesAsLiveData returns a LiveData with a list of OCShares`() {
+        val sharesLiveDataList: LiveData<List<OCShare>> = MutableLiveData(listOf(OC_SHARE))
+
         every {
-            remoteShareDataSource.insert(
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any()
+            localShareDataSource.getSharesAsLiveData(
+                filePath = filePath,
+                accountName = OC_ACCOUNT_NAME,
+                shareTypes = listOf(ShareType.PUBLIC_LINK, ShareType.USER, ShareType.GROUP, ShareType.FEDERATED)
             )
-        } returns shares[2]
+        } returns sharesLiveDataList
 
-        ocShareRepository.insertPrivateShare(
-            filePath,
-            ShareType.GROUP,
-            "whoever",
-            -1,
-            accountName
-        )
+        val sharesResult = ocShareRepository.getSharesAsLiveData(filePath, OC_ACCOUNT_NAME).getLastEmittedValue()
+        assertEquals(sharesLiveDataList.value, sharesResult)
```

Comment:
Well but here you are comparing again a wrapper (the list in this case), we want to compare the share object itself and that the list has a size of 1 (the only item there is the one we are comparing).

Alos, better to compare:
```suggestion
        assertEquals(OC_SHARE, sharesResult)
```

Reply:
Done!